worlds/osrs/__init__.py

```diff
@@ -0,0 +1,640 @@
+                                      (state.can_reach_region(RegionNames.Edgeville)
+                                       and woodcutting_rule_d1(state) and self.options.max_woodcutting_level >= 12)
+                    self.multiworld.register_indirect_condition(
+                        self.multiworld.get_region(RegionNames.Lumbridge, self.player), entrance)
+                    self.multiworld.register_indirect_condition(
+                        self.multiworld.get_region(RegionNames.South_Of_Varrock, self.player), entrance)
+                    self.multiworld.register_indirect_condition(
+                        self.multiworld.get_region(RegionNames.Barbarian_Village, self.player), entrance)
+                    self.multiworld.register_indirect_condition(
+                        self.multiworld.get_region(RegionNames.Edgeville, self.player), entrance)
+                # Access to other chunks based on woodcutting settings
+                if outbound_region_name == RegionNames.Lumbridge:
+                    entrance.access_rule = lambda state: woodcutting_rule_all(state) \
+                                                         and self.options.max_woodcutting_level >= 57
+                if outbound_region_name == RegionNames.South_Of_Varrock:
+                    entrance.access_rule = lambda state: woodcutting_rule_d3(state) \
+                                                         and self.options.max_woodcutting_level >= 42
+                if outbound_region_name == RegionNames.Barbarian_Village:
+                    entrance.access_rule = lambda state: woodcutting_rule_d2(state) \
+                                                         and self.options.max_woodcutting_level >= 27
+                # Edgeville does not need to be checked, because it's already adjacent
+
+    def roll_locations(self):
+        locations_required = 0
+        generation_is_fake = hasattr(self.multiworld, "generation_is_fake")  # UT specific override
+        for item_row in item_rows:
+            locations_required += item_row.count
+
+        locations_added = 1  # At this point we've already added the starting area, so we start at 1 instead of 0
+
+        # Quests are always added
+        for i, location_row in enumerate(location_rows):
+            if location_row.category in ["quest", "points", "goal"]:
+                self.create_and_add_location(i)
+                if location_row.category == "quest":
+                    locations_added += 1
+
+        # Build up the weighted Task Pool
+        rnd = self.multiworld.per_slot_randoms[self.player]
+
+        # Start with the minimum general tasks
+        general_tasks = [task for task in self.locations_by_category["general"]]
+        if not self.options.progressive_tasks:
+            rnd.shuffle(general_tasks)
+        else:
+            general_tasks.reverse()
+        for i in range(0, self.options.minimum_general_tasks):
```

Comment:
No need to specify 0 start
```suggestion
        for i in range(self.options.minimum_general_tasks):
```